pkgs/development/tools/detekt/default.nix

```diff
@@ -0,0 +1,28 @@
+{ detekt, lib, stdenv, fetchurl, makeWrapper, jre_headless, testers }:
+stdenv.mkDerivation rec {
+  pname = "detekt";
+  version = "1.22.0";
+
+  src = fetchurl {
+    url = "https://github.com/detekt/detekt/releases/download/v${version}/detekt-cli-${version}-all.jar";
+    sha256 = "sha256-NCOMBcAtk7cOlP3H8Bz/hfR/305j/DfaoFrwc504b/4=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    makeWrapper ${jre_headless}/bin/java $out/bin/${pname} --add-flags "-jar $src"
```

Comment:
the jar file should be copied from `$src` to `$out`, see examples like `pkgs/development/tools/database/trino-cli/default.nix`

Reply:
👍 Thanks for digging out the example, I'll make the change. 

For my own learning about Nix (I'm very much a newbie at the technology) - what's the difference between referencing `$src` and copying the jar to `$out`?

Reply:
Referencing $src would cause the built store path ($out) to depend on another store path ($src), and the reference is hidden inside the wrapper script, causing confusion especially when the users would like to do something with the jar (running it with a custom jre, etc.)